### PR TITLE
Projects: assign users to different versions of the projects table

### PR DESCRIPTION
**Pull request by Erin007**
The initial version of the React projects table has the (un)publish actions in the chevron dropdown menu, which potentially will be less discoverable than the button directly in the table as is the case in the current Angular table.  As we would like to promote publishing projects, we're going to run an A/B in which half of the users see the table with (un)publish in the chevron dropdown and the other half see (un)publish buttons directly in the table column. Then we we'll compare publish rates for each scenario. 

This PR assigns users based on whether they've been randomized into the user based experiment or not to different variants of the projects table (which can be viewed in the description of #24257).

screen shot 2018-08-15 at 1 40 34 pm

### CODE REVIEW

File: apps/src/sites/studio/pages/projects/index.js [hunk 1]
```diff
@@ -67,26 +71,54 @@ $(document).ready(() => {
...
-          <PersonalProjectsTable
-            canShare={projectsData.canShare}
-            publishMethod="chevron"
-          />
-        </Provider>,
-       document.getElementById('react-my-projects')
-      );
-    });
+
+    if (experiments.isEnabled(experiments.CHEVRON_PUBLISH_EXPERIMENT)) {
```
Comment (Erin007) [inline apps/src/sites/studio/pages/projects/index.js:82]: @balderdash this is the line I'm unsure about - how do I check if the user is in the publish-chevron experiment or not?

**Comment by Erin007** [comment]
@balderdash based on our conversation yesterday, I think I set up a user based experiment (see the PR description for what I ran).  Is this more like what you were expecting?

File: apps/src/sites/studio/pages/projects/index.js (hunk 1 above)
Comment (balderdash) [reply apps/src/sites/studio/pages/projects/index.js:82]: This is how you do it!

File: dashboard/app/views/projects/index.html.haml [hunk 2]
```diff
@@ -8,6 +8,12 @@
   // TODO Erin B - remove when we have enough featured Applab and Gamelab projects and/or profanity filter in place.
   - projects_data[:projectValidator] = current_user.permission?  UserPermission::PROJECT_VALIDATOR
   - projects_data[:canShare] = !current_user.sharing_disabled?
+  // We're going to run an A/B experiment to compare (un)publishing from the
+  // quick actions dropdown and from a button in the published column.
+  // As part of the experiment, we want to log users id.
+  // TODO (Erin B.) delete this data when we
+  // determine the experiment outcome.
+  - projects_data[:usedId] = current_user.id
```
Comment (balderdash) [inline dashboard/app/views/projects/index.html.haml]: Doesn't look like you need this anymore.

Comment (balderdash) [reply dashboard/app/views/projects/index.html.haml]: Oh I see, you're using it in https://github.com/code-dot-org/code-dot-org/pull/24322/files. I think you meant `userId`, not `usedId` though.

Comment (Erin007) [reply dashboard/app/views/projects/index.html.haml]: yes, good catch!

**Pull request merged**